googlesource/googlesource.go

```diff
@@ -38,44 +37,40 @@ var (
 type readRowInput struct {
 	tableID   string
 	offset    string
-	positions map[string]string
+	positions string
 	wg        *sync.WaitGroup
 }
 
 // checkInitialPos helps in creating the query to fetch data from endpoint
-func (s *Source) checkInitialPos(positions map[string]string, incrementColName map[string]string, tableID string, primaryColName map[string]string) (firstSync, userDefinedOffset bool, userDefinedKey bool) {
+func (s *Source) checkInitialPos(positions string, incrementColName string, tableID string, primaryColName string) (firstSync, userDefinedOffset bool, userDefinedKey bool) {
 	// if its the firstSync no offset is applied
-	if _, ok := positions[tableID]; !ok {
+	if positions == "" {
 		firstSync = true
 	}
 
 	// if incrementColName set - we orderBy the provided column name
-	if _, ok := incrementColName[tableID]; ok {
+	if len(incrementColName) > 0 {
 		userDefinedOffset = true
 	}
 
 	// if primaryColName set - we orderBy the provided column name
-	if _, ok := primaryColName[tableID]; ok {
+	if len(primaryColName) > 0 {
 		userDefinedKey = true
 	}
 
 	return firstSync, userDefinedOffset, userDefinedKey
 }
 
-// haris: why does rowInput need to be a chan?
-// Neha: the function is getting called inside a goroutine we get wrong value (everytime the last possible values) and
-// func param will change for each function call
-func (s *Source) ReadGoogleRow(rowInput chan readRowInput, responseCh chan sdk.Record) (err error) {
+func (s *Source) ReadGoogleRow(rowInput readRowInput, responseCh chan sdk.Record) (err error) {
```

Comment:
We should have a comment explaining what does this method do. Also, rowInput and responseCh are actually taken from the source itself. E.g. responeSh is always s.records. It looks like this method doesn't need any parameters.